flask_dance/contrib/salesforce.py

```diff
@@ -62,15 +64,25 @@ def make_salesforce_blueprint(
     if reprompt_consent:
         authorization_url_params["prompt"] = "consent"
 
+    subdomain = "test" if is_sandbox else "login"
+
     salesforce_bp = OAuth2ConsumerBlueprint(
         "salesforce",
         __name__,
         client_id=client_id,
         client_secret=client_secret,
         scope=scope,
-        base_url="https://login.salesforce.com/services/oauth2/",
-        authorization_url="https://login.salesforce.com/services/oauth2/authorize",
-        token_url="https://login.salesforce.com/services/oauth2/token",
+        base_url="https://{subdomain}.salesforce.com/services/oauth2/".format(
```

Comment:
Maybe this is a dumb question, but does Salesforce support any _other_ subdomains as well? Or only `test` and `login`?

Reply:
Salesforce has two types of environments: production and sandbox (testing). The authorization URLs (refreshing token URL as well) are different depending on the environment so production environments have `login` subdomain and sandbox environments have `test` one.

Reply:
I have checked it more precisely and your question was accurate. It is possible to create custom domains on the Salesforce so I am going to change that PR to handle any subdomain.